lib/rules/no-restricted-exports.js

```diff
@@ -109,7 +112,15 @@ module.exports = {
         function checkExportedName(node) {
             const name = astUtils.getModuleExportName(node);
 
-            if (restrictedNames.has(name)) {
+            let hasRestrictedNamePattern = false;
```

Comment:
Can we rename this to `matchesRestrictedNamePattern`? I think it would be more clear what true/false means.